doc/compiler.md

```diff
@@ -0,0 +1,205 @@
+# Using the ProtoDef compiler
```

Comment:
could you mention the compiler to the readme + add its API to API.md ?

Reply:
done